src/actions/posts.ts

```diff
@@ -196,79 +196,78 @@ export function createPost(post: Post, files: any[] = []) {
                 file_ids: fileIds,
             };
 
-            dispatch({
+            actions.push({
                 type: FileTypes.RECEIVED_FILES_FOR_POST,
                 postId: pendingPostId,
                 data: files,
             });
         }
 
-        dispatch({
+        actions.push({
             type: PostTypes.RECEIVED_NEW_POST,
             data: {
                 ...newPost,
                 id: pendingPostId,
             },
-            meta: {
-                offline: {
-                    effect: () => Client4.createPost({...newPost, create_at: 0}),
-                    commit: (result: any, payload: any) => {
-                        const actions: Action[] = [
-                            receivedPost(payload),
-                            {
-                                type: PostTypes.CREATE_POST_SUCCESS,
-                            },
-                            {
-                                type: ChannelTypes.INCREMENT_TOTAL_MSG_COUNT,
-                                data: {
-                                    channelId: newPost.channel_id,
-                                    amount: 1,
-                                },
-                            },
-                            {
-                                type: ChannelTypes.DECREMENT_UNREAD_MSG_COUNT,
-                                data: {
-                                    channelId: newPost.channel_id,
-                                    amount: 1,
-                                },
-                            },
-                        ];
-
-                        if (files) {
-                            actions.push({
-                                type: FileTypes.RECEIVED_FILES_FOR_POST,
-                                postId: payload.id,
-                                data: files,
-                            });
-                        }
-
-                        dispatch(batchActions(actions));
+        });
+
+        dispatch(batchActions(actions, 'BATCH_CREATE_POST_INIT'));
```

Comment:
Why give a name to the batched action? Is it to make debugging easier?

Reply:
I admit I copied this directly from the corresponding mobile repository, but yes, it does enhance the experience when debugging via Redux DevTools:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/1023171/110385463-586b9680-8035-11eb-9176-7d88e1ffa7d2.png">

Reply:
I wonder why it's "INIT" then. Regardless, that looks fine since it's probably better than seeing dozens of BATCH_ACTION in the dev tools

Reply:
In this case, it's to distinguish from the `BATCH_CREATE_POST` batch action a few lines later (and `BATCH_CREATE_POST_FAILED` later on). 0/5 on the naming.